webapp/distribution/mixins.py

```diff
@@ -0,0 +1,17 @@
+from braces.views._access import AccessMixin
+from django.core.exceptions import PermissionDenied
+
+
+class UserIsInvolvedMixin(AccessMixin):
+    def dispatch(self, request, *args, **kwargs):
+        shift = self.get_object()
```

Comment:
The logic looks fine but this mixin is only compatible to Shift related views, so maybe change the name.